src/main/java/de/tum/in/www1/artemis/web/rest/dto/metrics/CompetencyProgressDTO.java

```diff
@@ -0,0 +1,7 @@
+package de.tum.in.www1.artemis.web.rest.dto.metrics;
+
+/**
+ * A DTO representing a students progress for a competency.
+ */
+public record CompetencyProgressDTO(long competencyId, double progress, double confidence) {
```

Comment:
Could we add the annotation

@JsonInclude(JsonInclude.Include.NON_EMPTY)

here? We don't necessarily need it (yet), but it might be better to have it for the future

Reply:
We only want to add the annotation, once the DTO contains collections or other nullable objects.